client/actions/dashboard-action-creators.js

```diff
@@ -1,31 +1,39 @@
 import * as streetsApi from 'api/streets-api';
 import dashboardActionTypes from './dashboard-action-types';
 
-export function dashboardSearchSubmitted(streetObject) {
+export function dashboardSearchSubmitted(selectedStreet) {
     return dispatch => {
         dispatch({
-            type: dashboardActionTypes.SUBMITTED,
-            data: { streetObject },
+            type: dashboardActionTypes.SEARCH_SUBMITTED,
+            data: { selectedStreet },
         });
 
-        streetsApi.getStreet(streetObject.placeId)
+        streetsApi.getStreet(selectedStreet.place_id)
             .then(
-                response => dispatch(dashboardSearchSucceeded(response)),
-                error => dispatch(dashboardSearchFailed(error)),
+                response => dispatch(dashboardSearchSucceeded(selectedStreet, response)),
+                error => dispatch(dashboardSearchFailed(selectedStreet, error)),
             );
     };
 }
 
-export function dashboardSearchSucceeded(streetObject) {
+export function dashboardSearchSucceeded(selectedStreet, response) {
+
+    if (response) {
```

Comment:
This conditional check is unnecessary. You can just pass both `response` and `selectedStreet` to the `data` object and then you can remove the second return block.